diff --git a/plugins/FeedsPlugin.inc b/plugins/FeedsPlugin.inc
index fa1746b..08201d7 100644
--- a/plugins/FeedsPlugin.inc
+++ b/plugins/FeedsPlugin.inc
@@ -80,7 +80,7 @@ abstract class FeedsPlugin extends FeedsConfigurable implements FeedsSourceInter
   public function sourceDelete(FeedsSource $source) {}
 
   /**
-   * Loads on-behalf implementations from mappers/ directory.
+   * Loads on-behalf implementations from mappers/ directory and [modulename].feeds.inc files.
    *
    * FeedsProcessor::map() does not load from mappers/ as only node and user
    * processor ship with on-behalf implementations.
@@ -100,6 +100,11 @@ abstract class FeedsPlugin extends FeedsConfigurable implements FeedsSourceInter
           require_once("./$file->uri");
         }
       }
+      // Load [modulename].feeds.inc files.
+      $files = drupal_system_listing('/.*feeds\.inc$/', 'modules');
+      foreach ($files as $file) {
+        require_once("./$file->uri");
+      }
     }
     $loaded = TRUE;
   }
