Index: addressfield.module
===================================================================
--- addressfield.module	(revision 267)
+++ addressfield.module	(working copy)
@@ -5,6 +5,10 @@
  * Defines a field for attaching country-specific addresses to entities.
  */
 
+if (!module_exists('feeds')) {
+  module_load_include('inc', 'addressfield', 'addressfield.feeds');
+}
+
 /**
  * Implements hook_ctools_plugin_directory().
  */
